sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala

```diff
@@ -444,6 +445,26 @@ trait StreamTest extends QueryTest with SharedSQLContext with TimeLimits with Be
         }
       }
 
+      if (currentStream.isInstanceOf[MicroBatchExecution]) {
+        // Verify if stateful operators have correct metadata and distribution
+        // This can often catch hard to debug errors when developing stateful operators
+        val executedPlan = currentStream.lastExecution.executedPlan
+        executedPlan.collect { case s: StatefulOperator => s }.foreach { s =>
+          assert(s.stateInfo.isDefined)
+          assert(s.stateInfo.get.numPartitions >= 1)
+
+          s.requiredChildDistribution.foreach { d =>
+            withClue(s"$s specifies incorrect # partitions in requiredChildDistribution $d") {
+              assert(d.requiredNumPartitions.isDefined)
+              assert(d.requiredNumPartitions.get >= 1)
+              if (d != AllTuples) {
+                assert(d.requiredNumPartitions.get == s.stateInfo.get.numPartitions)
```

Comment:
can you also verify that this is equal to the number of partitions in the metadata?